DEVELOPING_OPAL/validate/src/test/scala/org/opalj/fpcf/StringAnalysisTest.scala

```diff
@@ -0,0 +1,252 @@
+/* BSD 2-Clause License - see OPAL/LICENSE for details. */
+package org.opalj
+package fpcf
+
+import java.io.File
+import java.net.URL
+
+import scala.collection.mutable.ListBuffer
+
+import org.opalj.collection.immutable.Chain
+import org.opalj.br.analyses.Project
+import org.opalj.br.Annotation
+import org.opalj.br.Method
+import org.opalj.br.cfg.CFG
+import org.opalj.br.Annotations
+import org.opalj.br.fpcf.properties.StringConstancyProperty
+import org.opalj.br.fpcf.FPCFAnalysesManagerKey
+import org.opalj.br.fpcf.FPCFAnalysis
+import org.opalj.br.fpcf.PropertyStoreKey
+import org.opalj.tac.Stmt
+import org.opalj.tac.TACStmts
+import org.opalj.tac.VirtualMethodCall
+import org.opalj.tac.fpcf.analyses.string_analysis.LazyInterproceduralStringAnalysis
+import org.opalj.tac.fpcf.analyses.string_analysis.LazyIntraproceduralStringAnalysis
+import org.opalj.tac.EagerDetachedTACAIKey
+import org.opalj.tac.cg.RTACallGraphKey
+import org.opalj.tac.fpcf.analyses.string_analysis.V
+
+/**
+ * @param fqTestMethodsClass The fully-qualified name of the class that contains the test methods.
+ * @param nameTestMethod The name of the method from which to extract DUVars to analyze.
+ * @param filesToLoad Necessary (test) files / classes to load. Note that this list should not
+ *                    include "StringDefinitions.class" as this class is loaded by default.
+ */
+sealed class StringAnalysisTestRunner(
+        val fqTestMethodsClass: String,
+        val nameTestMethod:     String,
+        val filesToLoad:        List[String]
+) extends PropertiesTest {
+
+    /**
+     * @return Returns all relevant project files (NOT including library files) to run the tests.
+     */
+    def getRelevantProjectFiles: Array[File] = {
+        val necessaryFiles = Array(
+            "properties/string_analysis/StringDefinitions.class"
+        ) ++ filesToLoad
+        val basePath = System.getProperty("user.dir")+
+            "/DEVELOPING_OPAL/validate/target/scala-2.12/test-classes/org/opalj/fpcf/"
```

Comment:
Having the fixed path here seems to be dangerous. Why is this needed at all?

Reply:
Fixed by revamping the testing classes